ui/page/dcrdex/market.go

```diff
@@ -0,0 +1,937 @@
+	switchLotsOrAmount *cryptomaterial.Switch
+	lotsAmountEditor   cryptomaterial.Editor
+	totalEditor        cryptomaterial.Editor
+
+	seeFullOrderBookBtn cryptomaterial.Button
+
+	createOrderBtn    cryptomaterial.Button
+	buyOrder          bool
+	immediateOrder    cryptomaterial.CheckBoxStyle
+	immediateMoreInfo *cryptomaterial.Clickable
+
+	openOrdersBtn       cryptomaterial.Button
+	orderHistoryBtn     cryptomaterial.Button
+	openOrdersDisplayed bool
+
+	ordersTableHorizontalScroll *widget.List
+
+	showLoader bool
+}
+
+func NewDEXMarketPage(l *load.Load) *DEXMarketPage {
+	th := l.Theme
+	pg := &DEXMarketPage{
+		Load:                        l,
+		ordersContainer:             &widget.List{List: layout.List{Axis: vertical, Alignment: layout.Middle}},
+		GenericPageModal:            app.NewGenericPageModal(DEXAccountOnboardingID),
+		scrollContainer:             &widget.List{List: layout.List{Axis: vertical, Alignment: layout.Middle}},
+		priceEditor:                 newTextEditor(l.Theme, values.String(values.StrPrice), "0", false),
+		totalEditor:                 newTextEditor(th, values.String(values.StrTotal), "", false),
+		switchLotsOrAmount:          l.Theme.Switch(), // TODO: Set last user choice, default is unchecked.
+		lotsAmountEditor:            newTextEditor(l.Theme, values.String(values.StrLots), "0", false),
+		createOrderBtn:              th.Button(values.String(values.StrBuy)), // TODO: toggle
+		buyOrder:                    true,
+		immediateOrder:              th.CheckBox(new(widget.Bool), values.String(values.StrImmediate)),
+		orderTypesDropdown:          th.DropDown(orderTypes, values.DEXOrderTypes, true),
+		immediateMoreInfo:           th.NewClickable(false),
+		addServerBtn:                th.NewClickable(false),
+		seeFullOrderBookBtn:         th.Button(values.String(values.StrSeeMore)),
+		toggleBuyAndSellBtn:         th.SegmentedControl([]string{values.String(values.StrBuy), values.String(values.StrSell)}),
+		openOrdersBtn:               th.Button(values.String(values.StrOpenOrders)),
+		orderHistoryBtn:             th.Button(values.String(values.StrTradeHistory)),
+		openOrdersDisplayed:         true,
+		ordersTableHorizontalScroll: &widget.List{List: layout.List{Axis: horizontal, Alignment: layout.Middle}},
+	}
+
+	btnPadding := layout.Inset{Top: dp8, Right: dp20, Left: dp20, Bottom: dp8}
+	pg.toggleBuyAndSellBtn.Padding = btnPadding
+	pg.openOrdersBtn.Inset, pg.orderHistoryBtn.Inset = btnPadding, btnPadding
+	pg.openOrdersBtn.Font.Weight, pg.orderHistoryBtn.Font.Weight = font.SemiBold, font.SemiBold
+
+	pg.orderTypesDropdown.SelectedItemDirectionAfterCollapse = layout.E
+	pg.orderTypesDropdown.Width = values.MarginPadding120
+	pg.orderTypesDropdown.FontWeight = font.SemiBold
+	pg.orderTypesDropdown.MakeSelectedItemHoverableAfterCollapse = false
+	pg.orderTypesDropdown.SelectedItemIconColor = &pg.Theme.Color.Primary
+	pg.orderTypesDropdown.ExpandedLayoutInset = layout.Inset{Top: values.MarginPadding30}
```

Comment:
Why? Share a screenshot highlighting how this top spacing is rendered.

Reply:
It ensures vertical alignment with the buySell btn since they are both stacked.

Reply:
> Why? Share a screenshot highlighting how this top spacing is rendered.

By default, `layout.Stack` stacks its children on top of each other. The last child is always at the top but the designs show the last child (which is the expanded view) below the collapsed layout and in another case (i.e the order type dropdown) the expanded view is just halfway below the collapsed layout.

NOTE: This extra top (margin) when expanded, eats space from the last stack child, hence the need to account for that with len(d.items) * item default height (dp50) + `d.ExpandedLayoutInset.Top`

<img width="736" alt="Screenshot 2023-11-28 at 9 08 58 PM" src="https://github.com/crypto-power/cryptopower/assets/57448127/b2967df7-f62d-4442-9498-d172509df46f">